File: algorithms/warmup/extra-long-factorials/extra-long-factorials.py
#!/usr/bin/python
import sys

def solve(n):
	p = 1
	for i in xrange(2, n + 1):
		p *= i
	return p

def load(stdin):
	N = int(stdin.readline())
	print solve(N)

if __name__ == '__main__':
	if len(sys.argv) == 2 and sys.argv[1] == "COHADAR":
		load(open("extra-long-factorials.in", "r"))
	else:
		load(sys.stdin)

File: algorithms/warmup/extra-long-factorials/ExtraLongFactorials.java
import java.util.*;
import java.io.*;
import java.math.*;

public class ExtraLongFactorials {

	BigInteger solve(int n) {
		BigInteger p = new BigInteger("1");
		for (int i = 2; i <= n; i++) {
			p = p.multiply(new BigInteger(String.valueOf(i)));
		}
		return p;
	}

	void load(Scanner scanner) {
		int N = scanner.nextInt();
		System.out.println(solve(N));
	}

	public static void main(String[] args) throws Exception {
		Scanner scanner;
		if (args.length == 1 && "COHADAR".equals(args[0])) {
			scanner = new Scanner(new File("extra-long-factorials.in"));
		} else {
			scanner = new Scanner(System.in);
		}
		new ExtraLongFactorials().load(scanner);
	}
}
